lib/msal-browser/src/broker/nativeBroker/PlatformAuthProvider.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    LoggerOptions,
+    IPerformanceClient,
+    Logger,
+    AuthenticationScheme,
+} from "@azure/msal-common/browser";
+import { name, version } from "../../packageMetadata.js";
+import {
+    BrowserConfiguration,
+    DEFAULT_NATIVE_BROKER_HANDSHAKE_TIMEOUT_MS,
+} from "../../config/Configuration.js";
+import { BrowserPerformanceClient } from "../../telemetry/BrowserPerformanceClient.js";
+import { PlatformAuthExtensionHandler } from "./PlatformAuthExtensionHandler.js";
+import { NativeConstants } from "../../utils/BrowserConstants.js";
+import { IPlatformAuthHandler } from "./IPlatformAuthHandler.js";
+import { PlatformAuthDOMHandler } from "./PlatformAuthDOMHandler.js";
+
+/**
+ * Checks if the platform broker is available in the current environment.
+ * @param loggerOptions
+ * @param perfClient
+ * @returns
+ */
+export async function isPlatformBrokerAvailable(
+    loggerOptions?: LoggerOptions,
+    perfClient?: IPerformanceClient
+): Promise<boolean> {
+    const logger = new Logger(loggerOptions || {}, name, version);
+
+    logger.trace("isPlatformBrokerAvailable called");
+
+    const defaultPerformanceClientConfig = {
+        auth: {
+            clientId: "",
+        },
+    };
+
+    const performanceClient =
+        perfClient ||
+        new BrowserPerformanceClient(defaultPerformanceClientConfig);
```

Comment:
nit: We have a default defined
```suggestion
        new StubPerformanceClient();
```